src/cli/commands/ignore.ts

```
@@ -14,16 +14,16 @@ import { MisconfiguredAuthInCI } from '../../lib/errors/misconfigured-auth-in-ci
 export default function ignore(options): Promise<MethodResult> {
   debug('snyk ignore called with options: %O', options);
 
-  return auth
-    .isAuthed()
-    .then((authed) => {
-      if (!authed) {
-        if (isCI()) {
-          throw MisconfiguredAuthInCI();
-        }
-      }
+  return new Promise(() => {
+    try {
       apiTokenExists();
-    })
+    } catch (err) {
+      if (isCI()) {
+        throw MisconfiguredAuthInCI();
+      }
+      throw err;
+    }
+  })
     .then(() => {
       return authorization.actionAllowed('cliIgnore', options);
     })
```

suggestion: this might need a catch in case the auth fails, the `actionAllowed` returns am object, that should return an auth error (AuthFailedError).

---

That is true before this change as well, right? I believe this to be handled by the thing using the promise 🤔

---

Yes. I was just thinking that the API call to check the `cliIgnoreAuthorization` seem to not care about the status codes (neither does the verify function call), which seems like a bad idea.

I would say this is the last thing to be taken care of for this PR, just to make sure there's some better error messages for the users. Happy to pair on this if you want to, feel free to book a time in my calendar.